RecoLocalTracker/SiStripRecHitConverter/interface/StripCPE.h

```diff
@@ -28,6 +29,45 @@ class StripCPE : public StripClusterParameterEstimator
 	    const SiStripLatency&);    
   LocalVector driftDirection(const StripGeomDetUnit* det) const;
 
+ struct Param {
+    Param() : topology(nullptr) {}
+    StripTopology const * topology;
+    LocalVector drift;
+    float thickness, invThickness,pitch_rel_err2, maxLength;
+    int nstrips;
+    float backplanecorrection;
+    SiStripDetId::ModuleGeometry moduleGeom;
+    float coveredStrips(const LocalVector&, const LocalPoint&) const;
+  };
+
+  
+  struct AlgoParam {
+    Param const & p; const LocalTrajectoryParameters & ltp;
+    SiStripDetId::SubDetector loc; float afullProjection; float corr;
+  };
+
+
+  virtual StripClusterParameterEstimator::LocalValues
+  localParameters( const SiStripCluster& cl, AlgoParam const & ap) const {
+    return std::make_pair(LocalPoint(), LocalError());
+  }
+  
+  AlgoParam getAlgoParam(const GeomDetUnit& det, const LocalTrajectoryParameters & ltp) const {
+
+    StripCPE::Param const & p = param(det);
+    SiStripDetId::SubDetector loc = SiStripDetId( det.geographicalId() ).subDetector();  
+ 
+    LocalVector track = ltp.momentum();
+    track *= -p.thickness/track.z();
```

Comment:
Is this division safe? Could `track.z()` be zero? We've just been fixing some bugs caused by an invalid TrajectoryStateOnSurface. There might be more.